### PR TITLE
Add Locations() method

### PR DESCRIPTION
**Pull request by sudoryan**

### CODE REVIEW

File: uns_service.go [hunk 1]
```diff
@@ -142,6 +147,82 @@ func (c *UnsService) httpUrl(domainName string) (string, error) {
...
+	dataOutstruct := GetDataForManyOutstruct{
+		Resolvers: []common.Address{},
+		Owners:    []common.Address{},
+		Values:    [][]string{},
+	}
+	data := ret[len(ret)-1]
+	err = proxyReaderMethodsAbi.UnpackIntoInterface(&dataOutstruct, "getDataForMany", data)
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (DeRain) [reply uns_service.go:210]: @DancingAxolotl Good idea! I think we will split this out once we will need to reuse this code.

File: uns.go [hunk 2]
```diff
@@ -140,6 +141,22 @@ func (c *Uns) DNS(domainName string, types []dnsrecords.Type) ([]dnsrecords.Reco
 	return nil, err
 }
 
+func (c *Uns) Locations(domainNames []string) (map[string]namingservice.Location, error) {
+	for _, domainName := range domainNames {
+		isSupported, _ := c.IsSupportedDomain((domainName))
```
Comment (DeRain) [inline uns.go:146]: Calling isSupportDomain for each domain name in array will introduce length(domainName) amount of network queries which makes multicall usage useless. If domain is not supported - don't return an error - just return empty location for this domain name.

File: uns_builder.go [hunk 3]
```diff
@@ -132,13 +132,17 @@ func (cb *unsBuilder) Build() (*Uns, error) {
 	if err != nil {
 		return nil, err
 	}
+	l1Service.networkId = NetworkNameToId[cb.l1Network]
```
Comment (DeRain) [inline uns_builder.go:135]: @sudoryan Just allow user to put networkId instead of network names

**Review by DeRain — CHANGES_REQUESTED** [review]

File: uns_builder.go (hunk 3 above)
Comment (sudoryan) [reply uns_builder.go:135]: unsBuilder.Build() currently takes in the network name. I don't think we should have the user pass in both the networkId and network name

Comment (sudoryan) [reply uns_builder.go:135]: So should we replace the mappings in uns_config.go from networkName to networkId?

**Pull request merged**